### PR TITLE
Sum fixes

### PR DESCRIPTION
**Pull request by markbates**

### CODE REVIEW

File: cmd/format-coverage.go [hunk 1]
```diff
@@ -38,26 +38,33 @@ func (f CoverageFormatter) Save() error {
...
-		return err
+		return errors.WithStack(err)
 	}
 
-	return rep.Save(out)
+	err = rep.Save(out)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	return nil
```
Comment (pbrisbin) [inline cmd/format-coverage.go]: The changes in this file feel un-related to sum-coverage. If so, could we break out their own commit and/or a separate PR?

File: cmd/sum-coverage.go [hunk 2]
```diff
@@ -66,6 +69,6 @@ var sumCoverageCmd = &cobra.Command{
 
 func init() {
 	sumCoverageCmd.Flags().BoolVarP(&summerOptions.Print, "print", "p", false, "prints to standard out only")
-	sumCoverageCmd.Flags().StringVarP(&summerOptions.Output, "output", "o", "codeclimate.json", "output path")
+	sumCoverageCmd.Flags().StringVarP(&summerOptions.Output, "output", "o", "coverage/codeclimate.json", "output path")
```
Comment (pbrisbin) [inline cmd/sum-coverage.go]: I'm feeling like the string `"coverage/codeclimate.json"` should be made a constant somewhere to avoid these kinds of bugs. This is, by my count, the 3rd place that needed changing.

File: formatters/source_file.go [hunk 3]
```diff
@@ -24,33 +25,42 @@ func (a SourceFile) Merge(b SourceFile) (SourceFile, error) {
...
+	lc := LineCounts{}
 
 	for _, c := range sf.Coverage {
-		if _, ok := c.(int); ok {
-			lc.Covered++
+		if !c.Valid {
+			continue
+		}
+		lc.Total++
+		if c.Int == 0 {
```
Comment (ale7714) [inline formatters/source_file.go]: If i understand correctly, the problem came from counting null as missed lines?

File: cmd/sum-coverage.go (hunk 2 above)
Comment (ale7714) [inline cmd/sum-coverage.go]: 👍

File: formatters/source_file.go (hunk 3 above)
Comment (pbrisbin) [inline formatters/source_file.go]: *cough* Haskell *cough*
